### ISSUE TITLE
Possible dates bug lib/plugins/processor/post.js

### ISSUE THREAD

**Issue by tcrowe** [hexojs/hexo#3397]
In this issue we found that if the user sets `timezone:` in their `_config.yml` the date is modified and can be different than what they expect.
https://github.com/hexojs/hexo/issues/3365

See this code here:
https://github.com/hexojs/hexo/blame/master/lib/plugins/processor/post.js#L78

https://github.com/hexojs/hexo/blame/90b107c8b0bd8b386f64463403391f67eb1b2535/lib/plugins/processor/post.ts#L123

`if timezone then change date`

I'm not certain if this is a bug or not so I need help to understand.

## How to test

1.

```sh
hexo init
npm install
hexo new post one
hexo new post two
hexo new post three
hexo serve
```

2. View markdown dates and then view post dates in browser

3. Set `timezone: Asia/Shanghai` inside `./_config.yml`

4. `hexo clean && hexo serve`

5. Repeat 2

---

If you do that you should notice the same date changes reported in #3365. Is this a bug or should hexo be changing the date relative to GMT or location?

Thanks for taking a look

**Comment by puremana** [comment]
Was just about to post an issue regarding this, glad to see it's probably because of my timezone (Pacific/Auckland).

A post made on the 19th of November is showing the post.date being 4th of November, all other dates are off too, although most are just off by one.

Here's a link to my repo - https://github.com/puremana/personal-blog
Affected post - https://github.com/puremana/personal-blog/blob/master/source/_posts/CORS-And-Hiding-Api-Keys.md

**Comment by YoshinoriN** [comment]
I'm not yet investigate, but #3282  is possibility that same problem.

**Comment by tcrowe** [comment]
Thank you, @YoshinoriN. I will search for all the places with timezone and see what else it's used for. Then I will make a PR to fix it. It's actually been there since the beginning. lol

**Comment by tcrowe** [comment]
I tried for a while but I did not come to a solution. I will try again tomorrow.

**Comment by puremana** [comment]
@tcrowe any luck?

**Comment by tcrowe** [comment]
No, @puremana. For now all we can do is ask people to disable(comment or delete) the `timezone:` configuration which will render the dates in the timezone of the machine generating the output source.

When I was looking at the code I couldn't figure out the original intention so I need to think about the implications.

Was it true that if you disable the `timezone:` that the dates show up as you expect?

**Comment by puremana** [comment]
Sorry, it turns out timezone doesn't change anything at all in my project, even when it should be. My bug was something else, thanks for all of your help

**Comment by stevenjoezhang** [comment]
Thank you for reporting this issue, I'm sorry it hasn’t been resolved over the past few years. I believe this issue shares the same root cause as https://github.com/hexojs/hexo/issues/3282 (explained in https://github.com/hexojs/hexo/issues/3282#issuecomment-462077102)

To consolidate the discussion, I will go ahead and close this issue. Please join the conversation in #3282. Thank you!
